File: src/main/scala/dao/ContentDao.scala
package dao

import model.Content
import util.HibernateUtil
import org.hibernate.Session
import org.hibernate.Transaction

import scala.collection.JavaConverters._

class ContentDao {

  def saveContent(content: Content): Unit = {
    var transaction: Transaction = null
    try {
      val session: Session = HibernateUtil.getSessionFactory.openSession()
      transaction = session.beginTransaction()
      session.save(content)
      transaction.commit()
    } catch {
      case e: Exception =>
        if (transaction != null) transaction.rollback()
        e.printStackTrace()
    }
  }

  def getAllContent: List[Content] = {
    val session: Session = HibernateUtil.getSessionFactory.openSession()
    session.createQuery("from Content", classOf[Content]).list().asScala.toList
  }

  def getContentById(id: Int): Content = {
    val session: Session = HibernateUtil.getSessionFactory.openSession()
    session.get(classOf[Content], id)
  }

  def updateContent(content: Content): Unit = {
    var transaction: Transaction = null
    try {
      val session: Session = HibernateUtil.getSessionFactory.openSession()
      transaction = session.beginTransaction()
      session.update(content)
      transaction.commit()
    } catch {
      case e: Exception =>
        if (transaction != null) transaction.rollback()
        e.printStackTrace()
    }
  }

  def deleteContent(id: Int): Unit = {
    var transaction: Transaction = null
    try {
      val session: Session = HibernateUtil.getSessionFactory.openSession()
      val content = session.get(classOf[Content], id)
      if (content != null) {
        transaction = session.beginTransaction()
        session.delete(content)
        transaction.commit()
      }
    } catch {
      case e: Exception =>
        if (transaction != null) transaction.rollback()
        e.printStackTrace()
    }
  }
}

File: src/main/java/dao/ContentDao.java
package dao;

import model.contentModel;
import org.hibernate.Session;
import org.hibernate.Transaction;
import org.hibernate.SessionFactory;
import org.hibernate.cfg.Configuration;

public class ContentDao {

    private SessionFactory factory;

    public ContentDao() {
        factory = new Configuration()
                .configure("hibernate.cfg.xml") // Ensure Hibernate config file is present
                .addAnnotatedClass(contentModel.class)
                .buildSessionFactory();
    }

    public void saveContent(contentModel content) {
        Session session = factory.getCurrentSession();

        try {
            // Start a transaction
            session.beginTransaction();

            // Save the content object
            session.save(content);

            // Commit the transaction
            session.getTransaction().commit();
        } finally {
            factory.close();
        }
    }

    // Add other CRUD operations (Update, Delete, Find by ID, etc.) as needed
}
